src/shop-detail.html

```diff
@@ -271,6 +303,16 @@
         });
       },
 
+      _buyNow: function() {
+        var item  = {
+          item: this.item,
+          quantity: parseInt(this.$.quantitySelect.value, 10),
+          size: this.$.sizeSelect.value
+        };
+
+        this.fire('buy-item', item);
```

Comment:
Unless I'm not seeing something in the event, `buy-item` needs the item included so that it can process the payment for that item in isolation. Otherwise we get into the business of cart manipulation for handling a single item, and I've opted to leave the cart alone for the "Buy Now" feature.